### ISSUE TITLE
[Bug]: Win32 links -ldbghelp before -labsl_symbolize

### ISSUE THREAD

**Issue by stanhu** [abseil/abseil-cpp#1497]
### Describe the issue

While attempting to cross-compile a `x86-mingw32` build (32-bit Windows on MinGW), I ran into this undefined reference to `SymFromAddr`, which comes from the Windows Debug Helper:

```
/usr/bin/i686-w64-mingw32-ld: /home/stanhu/github/re2/ports/i386-mingw32/i686-w64-mingw32/abseil/20230125.3/lib/libabsl_symbolize.a(symbolize.cc.obj):symbolize.cc:(.text+0x106): undefined reference to `_imp__SymFromAddr@20'
```

Looking closer, I see that `absl_symbolize.pc` incorrectly puts `-ldbghelp` before `-labsl_symbolize`:

```
prefix=/home/stanhu/github/clean/re2/ports/i386-mingw32/i686-w64-mingw32/abseil/20230125.3
exec_prefix=${prefix}
libdir=/home/stanhu/github/clean/re2/ports/i386-mingw32/i686-w64-mingw32/abseil/20230125.3/lib
includedir=/home/stanhu/github/clean/re2/ports/i386-mingw32/i686-w64-mingw32/abseil/20230125.3/include

Name: absl_symbolize
Description: Abseil symbolize library
URL: https://abseil.io/
Version: 20230125
Requires: absl_debugging_internal = 20230125, absl_demangle_internal = 20230125, absl_base = 20230125, absl_config = 20230125, absl_core_headers = 20230125, absl_dynamic_annotations = 20230125, absl_malloc_internal = 20230125, absl_raw_logging_internal = 20230125, absl_strings = 20230125
Libs: -L${libdir} -ldbghelp  -labsl_symbolize
Cflags: -I${includedir} -DNOMINMAX
```

This looks like it's generated by adding `PC_LINKOPTS` before the libraries: https://github.com/abseil/abseil-cpp/blob/8c466401a346d11b946b7b83cf6870d71383331c/CMake/AbseilHelpers.cmake#L214

I worked around the issue by adding `-ldbghelp` after `-labsl_symbolize`.

Is there an easy way to fix this order? @h-vetinari 

### Steps to reproduce the problem

Compile abseil-cpp on `x86-mingw32`. Easiest way to reproduce it:

```shell
git clone https://github.com/mudge/re2
cd re2
git checkout 98a4a9f5f55de0ed25931ffa155ae49c4c9d34b3.
bundle install
bundle exec rake gem:x86-mingw32
```

### What version of Abseil are you using?

c16a2f43206b0235d49d4f6155f285a4d4939c58

### What operating system and version are you using?

Linux Ubuntu 20.04

### What compiler and version are you using?

```
$ i686-w64-mingw32-g++ --version
i686-w64-mingw32-g++ (GCC) 9.3-posix 20200320
Copyright (C) 2019 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

### What build system are you using?

```
$ cmake --version
cmake version 3.16.3

CMake suite maintained and supported by Kitware (kitware.com/cmake).
```

**Comment by h-vetinari** [comment]
> This looks like it's generated by adding `PC_LINKOPTS` before the libraries:
> 
> https://github.com/abseil/abseil-cpp/blob/8c466401a346d11b946b7b83cf6870d71383331c/CMake/AbseilHelpers.cmake#L214
>
> I worked around the issue by adding `-ldbghelp` after `-labsl_symbolize`.
> 
> Is there an easy way to fix this order?

I'm not a maintainer here, but it looks to me like you've found the right place where this could be changed. I'd try to just put `PC_LINKOPTS` after the lib portions in the generator for the pkgconfig files. Good luck!

**Comment by stanhu** [comment]
>  I'd try to just put PC_LINKOPTS after the lib portions in the generator for the pkgconfig files. 

Thanks, I considered that but I worried that might create a link order issue in the other way. But in looking at the `.pc` files, it seems that the main linker flags that precede the ABSL linking are:

* all: `-L${libdir}`
*  `absl_base.pc`: `-ladvapi32`
* `absl_random_internal_seed_material.pc`: `-lbcrypt`
* `absl_symbolize.pc`: `-ldbghelp`

It seems to me in all these cases these libraries should go after the abseil libraries. I've submitted #1498.
